console_conf/controllers/identity.py

```diff
@@ -52,22 +53,11 @@ def get_core_version():
 
 def get_device_owner():
     """ Check if device is owned """
-
-    # TODO: use proper snap APIs.
-    try:
-        extrausers_fp = open('/var/lib/extrausers/passwd', 'r')
-    except FileNotFoundError:
-        return None
-    with extrausers_fp:
-        passwd_line = extrausers_fp.readline()
-        if passwd_line and len(passwd_line) > 0:
-            passwd = passwd_line.split(':')
-            result = {
-                'realname': passwd[4].split(',')[0],
-                'username': passwd[0],
-                'homedir': passwd[5],
-                }
-            return result
+    con = SnapdConnection('', '/run/snapd.socket')
+    for user in con.get('v2/users').json()['result']:
+        user = user['email'].split('@')[0]
+        if os.path.isdir('/home/' + user):
+            return user
```

Comment:
It should check if device is managed, then iterate users.

As users may exist, whilst not actually be owners. Thus this should not return a user, when device is unmanged, but has lxd user.

Reply:
Isn't the point of checking that /home/$user exists to exclude users like lxd? I assumed that that's what that was for...

Reply:
Even if the device is managed, the lxd user could still be present and still needs to be excluded...

Reply:
If the device is not managed, we should not be skipping the identity page.

And inverse, if device is managed, and there are no users, we must skip identity page and do something else.... but ideally console-conf is prevented from running due to shell code.

And yes, non managing users may have a home directory. The idea of being able to create "user-accounts".

This does duplicate the shell code, but because shell code is supposedly optimization path.

Reply:
I think you're right though that to match the desired behaviour in the bug, this bit does need to be fixed.